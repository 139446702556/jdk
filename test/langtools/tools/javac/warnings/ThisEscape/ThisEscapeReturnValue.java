/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @compile/ref=ThisEscapeReturnValue.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeReturnValue.java
 * @summary Verify 'this' escape detection properly handles leaks via method return values
 */

public class ThisEscapeReturnValue {

    public ThisEscapeReturnValue() {
        final Object rval = ThisEscapeReturnValue.method(this);
        ((ThisEscapeReturnValue)rval).mightLeak();
    }

    public static Object method(Object obj) {
        return obj;
    }

    public void mightLeak() {
    }
}
