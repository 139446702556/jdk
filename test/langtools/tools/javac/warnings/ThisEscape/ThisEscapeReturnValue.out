ThisEscapeReturnValue.java:12:48: compiler.warn.possible.this.escape
1 warning
